tests/props/host/host.c

```diff
@@ -8,8 +8,51 @@
 #include <openenclave/internal/tests.h>
 #include <stdio.h>
 #include "../../../host/sgx/enclave.h"
+#include "../host/sgx/cpuid.h"
 #include "props_u.h"
 
+#ifdef oe_get_xfrm
+#undef oe_get_xfrm
+#endif
+
+bool avx_supported()
+{
+    uint32_t eax, ebx, ecx, edx;
+
+    eax = ebx = ecx = edx = 0;
+
+    // Obtain feature information using CPUID Leaf 1
+    oe_get_cpuid(1, 0, &eax, &ebx, &ecx, &edx);
+
+    // Check if AVX instruction extensions (bit 28) are supported in the
+    // processor
+    if (!(ecx & (1 << 28)))
+        return false;
+    else
+        return true;
+}
+
+/* Set the Enclave XFRM to Legacy mode */
+static uint64_t cur_enclave_xfrm = SGX_XFRM_LEGACY;
+
+/*
+ * Overriding oe_get_xfrm() function in the core to facilitate iterating through
+ * Legacy and AVX configurations
+ */
+uint64_t oe_get_xfrm()
+{
+    return cur_enclave_xfrm;
+}
+
+static void _set_xfrm(uint64_t xfrm)
```

Comment:
Please group with local static functions at beginning of file.

Reply:
Done.